src/components/ReplaceServerDetails.js

```diff
@@ -66,22 +65,81 @@ class ReplaceServerDetails extends Component {
       });
   }
 
+  initInputs = () => {
+    let inputs = {};
+
+    if (this.isCompute()) {
+      MODEL_SERVER_PROPS.forEach(input_name => {
+        inputs[input_name] = '';
+      });
+    }
+    else {
+      REPLACE_SERVER_MAC_IPMI_PROPS.forEach(input_name => {
+        inputs[input_name] = '';
+      });
+    }
+
+    return Map(inputs);
+  }
+
+  initInputsValid = () => {
+    let inputValid = {};
+
+    if (this.isCompute()) {
+      MODEL_SERVER_PROPS.forEach(input_name => {
+        inputValid[input_name] = undefined;
+      });
+    }
+    else {
+      REPLACE_SERVER_MAC_IPMI_PROPS.forEach(input_name => {
+        inputValid[input_name] = undefined;
+      });
+    }
+
+    return Map(inputValid);
+  }
+
+  isServerInputsValid = () => {
+    // if it is compute node and install os is not checked
+    // only check MAC and IMPI inputs when user inputs them
+    if (this.isCompute() && !this.state.isInstallOsSelected) {
+      return this.state.isValid.every((value, key) =>{
+        if (REPLACE_SERVER_MAC_IPMI_PROPS.includes(key)) {
+          return value === true || value === undefined;
+        }
+        else {
+          return value === true;
+        }
+      });
+    }
+
+    return this.state.isValid.every((value) => value === true);
+  }
+
   isFormInputValid = () => {
 
     // The form is valid if all of the fields are valid.  The OS install password only has
     // to be valid when the OS Install checkbox is selected
-    return this.state.isValid.every((value) => value === true) &&
+    return this.isServerInputsValid() &&
       (this.state.isOsInstallPasswordValid || !this.state.isInstallOsSelected);
   }
 
   handleDone = () => {
```

Comment:
The function has some problems with naming and cohesion that I am really struggling to understand, and now that we are making substantive changes to this, I'd like to see it fixed if possible:
1. The name `data` is too generic to make any sense of, and it needs a better name. It it being passed to a function in the caller that is expecting a parameter of `server`, so I think server would be a much better name here.
2. `theProps` is similarly so generic that it is impossible to make sense out of. Worse yet, the caller seems to be using it in a variety of ways -- one function is looking for a `wipeDisk` field and ignoring the rest, while other code is doing other things. It is really difficult to understand how this is going to be populated, and it is not clear why all of these fields have to be combined together in ReplaceServerDetails into one big object. It would be far more easy to understand to call `this.props.doneAction` and explicitly list each parameter instead of combining them, i.e.
this.props.doneAction(server, this.state.isWipeDiskSelected, this.state.isInstallOsSelected, ...)
and then let the caller, `UpdateServers`, use them or package them according to its needs.

Reply:
will update